[2024-11-23T08:52:27.670+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-23T08:52:27.689+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ETL_usa_spending.ingest_usa_spending_data_task manual__2024-11-23T08:52:19.533263+00:00 [queued]>
[2024-11-23T08:52:27.700+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ETL_usa_spending.ingest_usa_spending_data_task manual__2024-11-23T08:52:19.533263+00:00 [queued]>
[2024-11-23T08:52:27.701+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2024-11-23T08:52:27.719+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): ingest_usa_spending_data_task> on 2024-11-23 08:52:19.533263+00:00
[2024-11-23T08:52:27.732+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=91371) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-11-23T08:52:27.734+0000] {standard_task_runner.py:72} INFO - Started process 91373 to run task
[2024-11-23T08:52:27.732+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'ETL_usa_spending', 'ingest_usa_spending_data_task', 'manual__2024-11-23T08:52:19.533263+00:00', '--job-id', '208', '--raw', '--subdir', 'DAGS_FOLDER/project_dag.py', '--cfg-path', '/tmp/tmpy6nc3_ky']
[2024-11-23T08:52:27.736+0000] {standard_task_runner.py:105} INFO - Job 208: Subtask ingest_usa_spending_data_task
[2024-11-23T08:52:27.790+0000] {task_command.py:467} INFO - Running <TaskInstance: ETL_usa_spending.ingest_usa_spending_data_task manual__2024-11-23T08:52:19.533263+00:00 [running]> on host 7da4b16cbed4
[2024-11-23T08:52:27.896+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='ETL_usa_spending' AIRFLOW_CTX_TASK_ID='ingest_usa_spending_data_task' AIRFLOW_CTX_EXECUTION_DATE='2024-11-23T08:52:19.533263+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-23T08:52:19.533263+00:00'
[2024-11-23T08:52:27.898+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-23T08:52:28.289+0000] {logging_mixin.py:190} INFO - Checking for new data to ingest into states
[2024-11-23T08:52:28.600+0000] {logging_mixin.py:190} WARNING - /opt/***/dags/helpers/db_utils.py:149 UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
[2024-11-23T08:52:28.695+0000] {logging_mixin.py:190} INFO - No new data to ingest for states
[2024-11-23T08:52:29.184+0000] {logging_mixin.py:190} INFO - File moved from s3://project-processed-data/usa_spending_data/states_20241123_083458.csv to s3://project-processed-data/archived/usa_spending_data/states_20241123_083458.csv
[2024-11-23T08:52:29.395+0000] {logging_mixin.py:190} INFO - Checking for new data to ingest into cfdaprograms
[2024-11-23T08:52:29.608+0000] {logging_mixin.py:190} WARNING - /opt/***/dags/helpers/db_utils.py:149 UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
[2024-11-23T08:52:29.672+0000] {logging_mixin.py:190} INFO - No new data to ingest for cfdaprograms
[2024-11-23T08:52:29.909+0000] {logging_mixin.py:190} INFO - File moved from s3://project-processed-data/usa_spending_data/cfda_20241123_083458.csv to s3://project-processed-data/archived/usa_spending_data/cfda_20241123_083458.csv
[2024-11-23T08:52:30.093+0000] {logging_mixin.py:190} INFO - Checking for new data to ingest into recipients
[2024-11-23T08:52:30.307+0000] {logging_mixin.py:190} WARNING - /opt/***/dags/helpers/db_utils.py:149 UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
[2024-11-23T08:52:30.402+0000] {logging_mixin.py:190} INFO - No new data to ingest for recipients
[2024-11-23T08:52:30.676+0000] {logging_mixin.py:190} INFO - File moved from s3://project-processed-data/usa_spending_data/recipients_20241123_083458.csv to s3://project-processed-data/archived/usa_spending_data/recipients_20241123_083458.csv
[2024-11-23T08:52:30.841+0000] {logging_mixin.py:190} INFO - Checking for new data to ingest into agencies
[2024-11-23T08:52:31.067+0000] {logging_mixin.py:190} WARNING - /opt/***/dags/helpers/db_utils.py:149 UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
[2024-11-23T08:52:31.131+0000] {logging_mixin.py:190} INFO - No new data to ingest for agencies
[2024-11-23T08:52:31.342+0000] {logging_mixin.py:190} INFO - File moved from s3://project-processed-data/usa_spending_data/agency_20241123_083458.csv to s3://project-processed-data/archived/usa_spending_data/agency_20241123_083458.csv
[2024-11-23T08:52:31.508+0000] {logging_mixin.py:190} INFO - Checking for new data to ingest into subagency
[2024-11-23T08:52:31.730+0000] {logging_mixin.py:190} WARNING - /opt/***/dags/helpers/db_utils.py:149 UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
[2024-11-23T08:52:31.792+0000] {logging_mixin.py:190} INFO - No new data to ingest for subagency
[2024-11-23T08:52:32.036+0000] {logging_mixin.py:190} INFO - File moved from s3://project-processed-data/usa_spending_data/sub_agency_20241123_083458.csv to s3://project-processed-data/archived/usa_spending_data/sub_agency_20241123_083458.csv
[2024-11-23T08:52:32.483+0000] {logging_mixin.py:190} INFO - Checking for new data to ingest into grants
[2024-11-23T08:52:32.714+0000] {logging_mixin.py:190} WARNING - /opt/***/dags/helpers/db_utils.py:149 UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
[2024-11-23T08:52:32.948+0000] {logging_mixin.py:190} INFO - No new data to ingest for grants
[2024-11-23T08:52:33.254+0000] {logging_mixin.py:190} INFO - File moved from s3://project-processed-data/usa_spending_data/grants_20241123_083458.csv to s3://project-processed-data/archived/usa_spending_data/grants_20241123_083458.csv
[2024-11-23T08:52:33.634+0000] {logging_mixin.py:190} INFO - Checking for new data to ingest into funding
[2024-11-23T08:52:33.891+0000] {logging_mixin.py:190} WARNING - /opt/***/dags/helpers/db_utils.py:149 UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
[2024-11-23T08:52:34.281+0000] {logging_mixin.py:190} INFO - No new data to ingest for funding
[2024-11-23T08:52:34.553+0000] {logging_mixin.py:190} INFO - File moved from s3://project-processed-data/usa_spending_data/funding_20241123_083458.csv to s3://project-processed-data/archived/usa_spending_data/funding_20241123_083458.csv
[2024-11-23T08:52:34.861+0000] {logging_mixin.py:190} INFO - Checking for new data to ingest into federalawards
[2024-11-23T08:52:35.069+0000] {logging_mixin.py:190} WARNING - /opt/***/dags/helpers/db_utils.py:149 UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
[2024-11-23T08:52:35.128+0000] {logging_mixin.py:190} INFO - Ingesting new data into federalawards
[2024-11-23T08:52:35.399+0000] {logging_mixin.py:190} INFO - Error: insert or update on table "federalawards" violates foreign key constraint "federalawards_recipient_id_fkey"
DETAIL:  Key (recipient_id)=(4e80d4c6-329c-27c3-5033-0ac2275b4cf4-C) is not present in table "recipients".
[2024-11-23T08:52:35.681+0000] {logging_mixin.py:190} INFO - Checking for new data to ingest into transactions
[2024-11-23T08:52:35.889+0000] {logging_mixin.py:190} WARNING - /opt/***/dags/helpers/db_utils.py:149 UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
[2024-11-23T08:52:36.114+0000] {logging_mixin.py:190} INFO - No new data to ingest for transactions
[2024-11-23T08:52:36.365+0000] {logging_mixin.py:190} INFO - File moved from s3://project-processed-data/usa_spending_data/transactions_20241123_083458.csv to s3://project-processed-data/archived/usa_spending_data/transactions_20241123_083458.csv
[2024-11-23T08:52:36.678+0000] {logging_mixin.py:190} INFO - Checking for new data to ingest into subawards
[2024-11-23T08:52:36.899+0000] {logging_mixin.py:190} WARNING - /opt/***/dags/helpers/db_utils.py:149 UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
[2024-11-23T08:52:37.102+0000] {logging_mixin.py:190} INFO - Ingesting new data into subawards
[2024-11-23T08:53:15.237+0000] {logging_mixin.py:190} INFO - Data successfully written to table subawards.
[2024-11-23T08:53:15.509+0000] {logging_mixin.py:190} INFO - File moved from s3://project-processed-data/usa_spending_data/sub_awards_20241123_083458.csv to s3://project-processed-data/archived/usa_spending_data/sub_awards_20241123_083458.csv
[2024-11-23T08:53:15.512+0000] {python.py:240} INFO - Done. Returned value was: None
[2024-11-23T08:53:15.525+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-23T08:53:15.526+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=ETL_usa_spending, task_id=ingest_usa_spending_data_task, run_id=manual__2024-11-23T08:52:19.533263+00:00, execution_date=20241123T085219, start_date=20241123T085227, end_date=20241123T085315
[2024-11-23T08:53:15.573+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2024-11-23T08:53:15.591+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-11-23T08:53:15.594+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
